pkg/controllers/inventory/inventory_controller.go

```diff
@@ -342,15 +343,24 @@ func (r *ReconcileBareMetalAsset) checkAssetSecret(ctx context.Context, instance
 	})
 
 	// Set BaremetalAsset instance as the owner and controller
+	modified := false
 	if secret.OwnerReferences == nil || len(secret.OwnerReferences) == 0 {
+		modified = true
 		if err := controllerutil.SetControllerReference(instance, secret, r.scheme); err != nil {
 			klog.Errorf("Failed to set ControllerReference, %v", err)
 			return err
 		}
-		if err := r.client.Update(ctx, secret); err != nil {
-			klog.Errorf("Failed to update secret with OwnerReferences, %v", err)
-			return err
-		}
+	}
+
+	secretTypeLabel := map[string]string{
+		"cluster.open-cluster-management.io/credentials": "",
```

Comment:
These are the labels used by the Credentials showing up in the Credentials UI

I am not sure which one , I think the second label, will result in showing the secret in the Credentials UI, as a bma type. Do you have such a Credentials UI template ? Do you want those secrets to show up under the Credentials page ? If you do the 2 labels are good..

For backup purpose though, you can use either of the following secrets, both result in backing up the secrets having any of this labels: 
- cluster.open-cluster-management.io/type
- cluster.open-cluster-management.io/backup
If the label is added just for the purpose of backing up the secret ( no UI interaction ), then I suggest to use cluster.open-cluster-management.io/backup